File: 704.二分查找.java

/*
 * @lc app=leetcode.cn id=704 lang=java
 *
 * [704] 二分查找
 */

// @lc code=start
class Solution {
    public int search(int[] nums, int target) {
        int min = 0;
        int max = nums.length-1;
        while (max-min>0){
            int mid = min+((max-min)/2);
            if (nums[mid] == target){
                return mid;
            }
            if (nums[mid]>target){
                max = mid-1;
            }
            if (nums[mid]<target){
                min = mid +1;
            }
        }
        return -1;

    }
}
// @lc code=end

File: 704.二分查找.go
package leetcode

/*
 * @lc app=leetcode.cn id=704 lang=golang
 *
 * [704] 二分查找
 */

// @lc code=start
func search(nums []int, target int) int {
	min := 0
	max := len(nums) - 1
	for i := 0; i < len(nums); i++ {
		if max-min <= 0 {
			break
		}
		mid := min + (max-min)/2
		if nums[mid] == target {
			return mid
		}
		if nums[mid] > target {
			max = mid - 1
		}
		if nums[mid] < target {
			min = mid + 1
		}
	}
	return -1
}

// @lc code=end
